### ISSUE TITLE
training process freeze suddenly

### ISSUE THREAD

**Issue by johnSmith1990** [Sanjana7395/Face_segmentation#4]
Hi dear 
Thank you for your project.
when I run train.py script, it start for training and prints these lines then go to pause and don't print anything.

`
      1/Unknown - 13s 13s/step - loss: 3.1658 - acc: 0.0616 - mIoU: 0.01282020-11-04 11:56:42.866834: I tensorflow/core/platform/default/device_tracer.cc:641] Collecting 1278 kernel records, 234 memcpy records.

      2/Unknown - 13s 7s/step - loss: 3.1030 - acc: 0.0730 - mIoU: 0.0139 
      3/Unknown - 14s 5s/step - loss: 3.0520 - acc: 0.0800 - mIoU: 0.0149
      4/Unknown - 14s 4s/step - loss: 3.0151 - acc: 0.0868 - mIoU: 0.0158
      5/Unknown - 15s 3s/step - loss: 2.9743 - acc: 0.0991 - mIoU: 0.0177
      6/Unknown - 15s 2s/step - loss: 2.9408 - acc: 0.1135 - mIoU: 0.0196
      7/Unknown - 15s 2s/step - loss: 2.9064 - acc: 0.1288 - mIoU: 0.0217
      8/Unknown - 16s 2s/step - loss: 2.8769 - acc: 0.1451 - mIoU: 0.0240
      9/Unknown - 16s 2s/step - loss: 2.8478 - acc: 0.1617 - mIoU: 0.0262
     10/Unknown - 16s 2s/step - loss: 2.8215 - acc: 0.1782 - mIoU: 0.0285
     11/Unknown - 17s 2s/step - loss: 2.7944 - acc: 0.1972 - mIoU: 0.0311
     12/Unknown - 17s 1s/step - loss: 2.7659 - acc: 0.2168 - mIoU: 0.0337
     13/Unknown - 18s 1s/step - loss: 2.7429 - acc: 0.2321 - mIoU: 0.0358
     14/Unknown - 18s 1s/step - loss: 2.7188 - acc: 0.2492 - mIoU: 0.0382
     15/Unknown - 18s 1s/step - loss: 2.6954 - acc: 0.2673 - mIoU: 0.0409
     16/Unknown - 19s 1s/step - loss: 2.6758 - acc: 0.2847 - mIoU: 0.0432
     17/Unknown - 19s 1s/step - loss: 2.6534 - acc: 0.3008 - mIoU: 0.0455
     18/Unknown - 19s 1s/step - loss: 2.6323 - acc: 0.3146 - mIoU: 0.0474
     19/Unknown - 20s 1s/step - loss: 2.6122 - acc: 0.3281 - mIoU: 0.0494
     20/Unknown - 20s 1s/step - loss: 2.5933 - acc: 0.3395 - mIoU: 0.0512
     21/Unknown - 21s 978ms/step - loss: 2.5727 - acc: 0.3521 - mIoU: 0.0529
     22/Unknown - 21s 951ms/step - loss: 2.5521 - acc: 0.3646 - mIoU: 0.0546
     23/Unknown - 21s 925ms/step - loss: 2.5342 - acc: 0.3750 - mIoU: 0.0561
     24/Unknown - 22s 902ms/step - loss: 2.5147 - acc: 0.3862 - mIoU: 0.0577
     25/Unknown - 22s 880ms/step - loss: 2.4969 - acc: 0.3958 - mIoU: 0.0589
     26/Unknown - 22s 861ms/step - loss: 2.4781 - acc: 0.4054 - mIoU: 0.0602
     27/Unknown - 23s 843ms/step - loss: 2.4590 - acc: 0.4152 - mIoU: 0.0617
     28/Unknown - 23s 826ms/step - loss: 2.4409 - acc: 0.4242 - mIoU: 0.0630
     29/Unknown - 23s 810ms/step - loss: 2.4240 - acc: 0.4320 - mIoU: 0.0640`

**Comment by Sanjana7395** [comment]
Hi, 
With the GPU: GeForce GTX 1080 (Driver Version: 440.100, CUDA Version: 10.2), I am able to train completely for 50 epochs. 

If you have adopted [multi-worker training with keras](https://www.tensorflow.org/tutorials/distribute/multi_worker_with_keras), it might be [this](https://discuss.pytorch.org/t/training-freezes-after-some-epochs-no-error/92647) issue. So remove it and try doing.
Else while creating batches of data in preprocessing/augment_data/get_data() set **num_parallel_calls = None** and try training.
Hope this helps.

**Comment by johnSmith1990** [comment]
thanks. I dont use multi-worker.
also I set num_parallel_calls = None but it doesnt help.
